### ISSUE TITLE
Some questions about the Pedestrian and Cyclist

### ISSUE THREAD

**Issue by 1843744321mark** [hailanyi/CasA#16]
Hello. I try to build our own algorithm in this baseline. I have trained all of the KITTI training set and submit the test results to the website of KITTI. The test results on Car categories are very good, but I find that our test results on the Pedestrian and Cyclist categories are worse. Can you give me some advice for these two categories or should I change some configs? Besides, which metrics should I employ to determine the optimal epoch for submission, and is it still advisable to use validation set metrics for evaluation?

**Comment by hailanyi** [comment]
Yes, the best submission config to the test set is different from val set, such as the score threshold. When you use all the training set as train set, the best model is generated by ensembling last 10 epochs. Sometimes, the TTA can be employed here to merge the results. Sometimes, model average can be employed to merge multiple model to a single model, and produce the final results.

**Comment by 1843744321mark** [comment]
Thanks for your reply! I have another question: I have trained all of the KITTI training set (training+validation) and validated them on the validation set. The metrics on the Car and Cyclist categories are both over 0.98, which is normal. But the metrics on Pedestrian are very low. It might mean that the model does not use all Pedestrian data for training.  Could you give me some advice to address this problem?
2023-04-14 10:55:08,465   INFO  Car AP@0.70, 0.70, 0.70:
bbox AP:98.7939, 96.8302, 97.2020
bev  AP:98.6346, 96.4877, 90.1290
3d   AP:98.7447, 89.9579, 89.7099
aos  AP:98.78, 96.76, 97.09
Car AP_R40@0.70, 0.70, 0.70:
bbox AP:99.5897, 98.3489, 98.2522
bev  AP:99.4942, 97.9267, 95.8176
3d   AP:99.3197, 95.4830, 95.1450
aos  AP:99.58, 98.29, 98.16
Car AP@0.70, 0.50, 0.50:
bbox AP:98.7939, 96.8302, 97.2020
bev  AP:98.7651, 96.7499, 97.1162
3d   AP:98.7603, 96.7158, 97.0855
aos  AP:98.78, 96.76, 97.09
Car AP_R40@0.70, 0.50, 0.50:
bbox AP:99.5897, 98.3489, 98.2522
bev  AP:99.5642, 98.2811, 98.2136
3d   AP:99.5628, 98.2552, 98.1839
aos  AP:99.58, 98.29, 98.16
Pedestrian AP@0.50, 0.50, 0.50:
bbox AP:82.2350, 79.8112, 74.9736
bev  AP:86.0770, 83.5242, 81.1547
3d   AP:85.5244, 82.4212, 80.6760
aos  AP:80.23, 77.46, 72.56
Pedestrian AP_R40@0.50, 0.50, 0.50:
bbox AP:83.2041, 80.3366, 77.8997
bev  AP:88.8364, 84.7489, 81.8132
3d   AP:88.2581, 83.9963, 81.1392
aos  AP:81.04, 77.77, 75.09
Pedestrian AP@0.50, 0.25, 0.25:
bbox AP:82.2350, 79.8112, 74.9736
bev  AP:87.0242, 85.3600, 83.9833
3d   AP:87.0242, 85.3600, 83.9833
aos  AP:80.23, 77.46, 72.56
Pedestrian AP_R40@0.50, 0.25, 0.25:
bbox AP:83.2041, 80.3366, 77.8997
bev  AP:91.3057, 89.0625, 86.4413
3d   AP:91.3051, 89.0660, 86.4266
aos  AP:81.04, 77.77, 75.09
Cyclist AP@0.50, 0.50, 0.50:
bbox AP:99.0480, 96.1513, 94.7467
bev  AP:98.5128, 94.5614, 93.0728
3d   AP:98.3378, 94.4346, 92.9358
aos  AP:99.01, 96.11, 94.71
Cyclist AP_R40@0.50, 0.50, 0.50:
bbox AP:99.1881, 98.2296, 97.2655
bev  AP:98.7268, 96.8685, 95.5138
3d   AP:98.5927, 96.7316, 95.3523
aos  AP:99.15, 98.20, 97.23
Cyclist AP@0.50, 0.25, 0.25:
bbox AP:99.0480, 96.1513, 94.7467
bev  AP:98.5679, 94.6138, 93.1330
3d   AP:98.5679, 94.6138, 93.1330
aos  AP:99.01, 96.11, 94.71
Cyclist AP_R40@0.50, 0.25, 0.25:
bbox AP:99.1881, 98.2296, 97.2655
bev  AP:98.7840, 96.9250, 95.6506
3d   AP:98.7840, 96.9250, 95.6506
aos  AP:99.15, 98.20, 97.23

**Comment by 1843744321mark** [comment]
@hailanyi Hello! I note that “During testing, we kept the top 100 proposals for cascade refinement, and finally use NMS IoU 0.1 to produce the final detection results” in the paper. However, this config sets 0.75 in this project.
TEST:
                NMS_TYPE: nms_gpu
                MULTI_CLASSES_NMS: False
                USE_FAST_NMS: True
                SCORE_THRESH: 0.0
                NMS_PRE_MAXSIZE: 4000
                NMS_POST_MAXSIZE: 100
                NMS_THRESH: 0.75
Should I modify it to 0.1?

**Comment by hukefy** [comment]
> Yes, the best submission config to the test set is different from val set, such as the score threshold. When you use all the training set as train set, the best model is generated by ensembling last 10 epochs. Sometimes, the TTA can be employed here to merge the results. Sometimes, model average can be employed to merge multiple model to a single model, and produce the final results.
Hello, I would like to know about model ensembl/average and TTA in your answer, could you please explain it specifically? Or are there any references? Looking forward to your reply, thank you!

**Comment by hukefy** [comment]
> Yes, the best submission config to the test set is different from val set, such as the score threshold. When you use all the training set as train set, the best model is generated by ensembling last 10 epochs. Sometimes, the TTA can be employed here to merge the results. Sometimes, model average can be employed to merge multiple model to a single model, and produce the final results.

@hailanyi Hello, I would like to know about model ensembl/average and TTA in your answer above, could you please explain it specifically? Or are there any references? Looking forward to your reply, thank you!
